### DISCUSSION TITLE
audio is breaking up

### DISCUSSION THREAD

**Discussion by albrecht-lindner** [pschatzmann/arduino-audio-tools#963] [Q&A] [answered]
Hi,

I'm using the example ["A Simple SdFat Audio Player"](https://github.com/pschatzmann/arduino-audio-tools/tree/main/examples/examples-player/player-sdfat-i2s).
In accordance with this [performance article](https://github.com/pschatzmann/arduino-audio-tools/wiki/Performance) I increased _DEFAULT_BUFFER_SIZE_ (2048), _I2S_BUFFER_SIZE_ (1024) and _I2S_BUFFER_COUNT_ (20). I also set the audio logger to _AudioLogger::Warning_.

My test mp3 files are normal: stereo format at 44.1 kHz with a duration of approximately 3 minutes and file sizes between 2Mb and 3Mb

Still, the audio is breaking up and I get plenty of logs saying _[E] Fade.h : 137 - channels=0_

Does anyone have ideas what else I can try?

**Comment by albrecht-lindner** [comment]
I guess it is helpful to add that I am using a "ESP32 devkit v1".

Thanks so much in advance.

**Comment by pschatzmann** [comment, marked as answer]
Try to find out why setAudioInfo() is not called automatically with the correct values from the decoder e.g. by setting the log level to Debug. The fading functionality needs to know how many channels it has to process and 0 channels is definitely not valid.

A work around would be to call setAudioInfo() to initialise the player with some valid values. 
Alternatively you can try to switch the automatic fade off by calling setAutoFade(false)

Are you sure that you installed the latest versions of all libraries. See https://github.com/pschatzmann/arduino-audio-tools/wiki/It's-not-working
